core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -172,6 +184,21 @@ private boolean addMember(Address joiner, CacheJoinInfo joinInfo) {
             return false;
          }
 
+         // Validate if the member is allowed to join
+         if (persistentState.isPresent()) {
```

Comment:
The coordinator won't give the ok until ALL required nodes are in the initial cluster. So if the original cluster had 4 nodes on graceful shutdown, and when restarting they form two two-nodes clusters, neither coordinator will allow the join to proceed. Once the two clusters merge, the new coordinator will have all the required nodes.